### PR TITLE
Refactor sync_default_gems

### PR DESCRIPTION
**Pull request by nobu**

### CODE REVIEW

File: tool/sync_default_gems.rb [hunk 1]
```diff
@@ -446,14 +446,17 @@ def ignore_file_pattern_for(gem)
...
     # Gem-specific patterns
     case gem
+    when "rubygems"
+      # We don't copy any vcr_cassettes to this repository. Because the directory does not
+      # exist, rename detection doesn't work. So it starts with the original path `bundler/`.
+      %r[\A(?:
+        bundler/spec/support/artifice/vcr_cassettes
+      )\z]mx
     when "yarp"
+      # Rename detection never works between ruby/ruby/doc and ruby/yarp/docs
```
Comment (k0kubun) [inline tool/sync_default_gems.rb:456]: Is this comment relevant to what we're doing here?

**Review by k0kubun — APPROVED** [review]

File: tool/sync_default_gems.rb (hunk 1 above)
Comment (nobu) [reply tool/sync_default_gems.rb:457]: Just moved as-is.
The rename detection seems working to me.
Something changed?

Comment (nobu) [reply tool/sync_default_gems.rb]: These gem-specific patterns seem no longer needed.

**Review by hsbt — APPROVED** [review]

File: tool/sync_default_gems.rb [hunk 2]
```diff
@@ -598,52 +549,73 @@ def resolve_conflicts(gem, sha, edit)
...
+    unless remove.empty?
+      puts "Remove added files: #{remove.join(', ')}"
+      system(*%w"git rm -fr --", *remove)
+      if picked
+        system(*%w"git commit --amend --no-edit --", *remove, %i[out err] => File::NULL)
+      end
     end
-    tools = changed.select {|f|f.start_with?("test/fixtures/", "test/lib/", "tool/")}
-    unless tools.empty?
-      system(*%W"git rm -r --", *tools)
```
Comment (k0kubun) [inline tool/sync_default_gems.rb:627]: Why did you remove this `git rm`? This doesn't seem to work (See #alerts-sync on Slack for logs).

File: tool/sync_default_gems.rb (hunk 1 above)
Comment (k0kubun) [reply tool/sync_default_gems.rb]: > These gem-specific patterns seem no longer needed.

They were necessary. I added them back in https://github.com/ruby/ruby/commit/43ab2acf8242c580b1ef5d644ea94d48669d4869.
